src/Service/Catalog/Product.php

```diff
@@ -171,6 +213,54 @@ public function update(MageProduct $product): void
     public function delete(MageProduct $product): void
     {
         $this->shouldSynchronize($product);
-        $this->utility->initSdk()->getCatalogService()->deleteProduct((string) $product->getId());
+
+        foreach ($this->aggregateProductIds($product) as $productId) {
+            $this->utility->initSdk()->getCatalogService()->deleteProduct((string) $productId);
```

Comment:
should be string, unless u are not a believer in ur code
```suggestion
            $this->utility->initSdk()->getCatalogService()->deleteProduct($productId);
```